### ISSUE TITLE
x/playground: "// GOEXPERIMENT=loopvar“ line often causes "timeout running go build”

### ISSUE THREAD

**Issue by go101** [golang/go#61985]
### What version of Go are you using (`go version`)?

<pre>
$ go version
1.21
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

https://go.dev/play/p/VwcNVnozJlu

### What did you expect to see?

```

```

### What did you see instead?

```
go: unknown GOEXPERIMENT loopvar

Go build failed.
```

**Comment by thediveo** [comment]
Seeing this too. Does opting into the experimentneed rebuilding parts of the stdlib, such as fmt?

**Comment by bcmills** [comment]
(attn @findleyr @golang/release)

**Comment by findleyr** [comment]
This is probably due to recompilation of std archives (which are cached following https://go.dev/cl/460635).

Perhaps we should add a special case and build archives with `GOEXPERIMENT=loopvar` here:
(https://cs.opensource.google/go/x/playground/+/master:Dockerfile;l=90;drc=6821b5629810defb1ea68149e090b5b86576d7ca)

@bcmills can you confirm whether adding a second build of `std` would suffice? (I'm not familiar with GOEXPERIMENT handling in the Go command).

CC @rsc, who added support for `GOEXPERIMENT` in https://go.dev/cl/511640.

**Comment by bcmills** [comment]
> can you confirm whether adding a second build of std would suffice? (I'm not familiar with GOEXPERIMENT handling in the Go command).

Yes, adding `GOCACHE=/gocache GOEXPERIMENT=loopvar ./bin/go install --tags=faketime std` should suffice.

That said, I question whether the benefit of doing so would be worth the cost, given the need to deploy and maintain it. I hope that the revised loop variable semantics will soon be available without setting `GOEXPERIMENT` at all. 🙃

**Comment by findleyr** [comment]
@fzipp that's a great point.  In that case, I don't think we should bother making this work better for 1.21. It would increase the disk space required for every build, and as @bcmills noted would require maintenance.

**Comment by DeedleFake** [comment]
With 1.22 out, this seems to be even worse with `GOEXPERIMENT=rangefunc`. I'm getting timeouts 100% of the time now with [this code](https://go.dev/play/p/tZ9vT9Sh7eg).

**Comment by dolmen** [comment]
In #73449 I provide a curl one-liner to reproduce the issue.

GOEXPERIMENT are currently unusable on the Go Playground due to this build timeout.

**Comment by phuclv90** [comment]
`GOEXPERIMENT=simd` also doesn't work on the playground
```
// GOEXPERIMENT=simd
package main

import "fmt"
func main() {
	fmt.Println()
}
```
